src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        self.assertEqual(self.MPPC.cellDigitalGain.value,
+                         tuple(tuple(0.0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
+
+        # Test int as type (should be converted)
+        self.MPPC.cellDigitalGain.value = tuple(
+                tuple(int(0) for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1]))
+        self.assertEqual(self.MPPC.cellDigitalGain.value,
+                         tuple(tuple(0.0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
+
+        # Test invalid type
+        with self.assertRaises(ValueError):
+            self.MPPC.cellDigitalGain.value = tuple(tuple('string_type' for i in range(0, self.MPPC.shape[0]))
+                                                    for i in range(0, self.MPPC.shape[1]))
+        self.assertEqual(self.MPPC.cellDigitalGain.value,
+                         tuple(tuple(0.0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
+
+        # Test minimum value setter
+        with self.assertRaises(ValueError):
+            self.MPPC.cellDigitalGain.value = tuple(tuple(-1.0 for i in range(0, self.MPPC.shape[0]))
+                                                    for i in range(0, self.MPPC.shape[1]))
+        self.assertEqual(self.MPPC.cellDigitalGain.value,
+                         tuple(tuple(0.0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
+
+    def test_cellCompleteResolutionVA(self):
+        min_res = self.MPPC.cellCompleteResolution.range[0][0]
+        max_res = self.MPPC.cellCompleteResolution.range[1][0]
+
+        # Check if small resolution values are allowed
+        self.MPPC.cellCompleteResolution.value = (min_res + 5, min_res + 5)
+        self.assertEqual(self.MPPC.cellCompleteResolution.value, (min_res + 5, min_res + 5))
+
+        # Check if big resolutions values are allowed
+        self.MPPC.cellCompleteResolution.value = (max_res - 200, max_res - 200)
+        self.assertEqual(self.MPPC.cellCompleteResolution.value, (max_res - 200, max_res - 200))
+
+        # Check if VA refuses to set limits outside allowed range
+        with self.assertRaises(IndexError):
+            self.MPPC.cellCompleteResolution.value = (max_res + 10, max_res + 10)
+
+        self.assertEqual(self.MPPC.cellCompleteResolution.value, (max_res - 200, max_res - 200))
+
+        with self.assertRaises(IndexError):
+            self.MPPC.cellCompleteResolution.value = (min_res - 1, min_res - 1)
+        self.assertEqual(self.MPPC.cellCompleteResolution.value, (max_res - 200, max_res - 200))
+
+        # Check if setter prevents settings of non-square resolutions
+        self.MPPC.cellCompleteResolution.value = (int(0.2 * max_res), int(0.5 * max_res))
+        self.assertEqual(self.MPPC.cellCompleteResolution.value, (int(0.2 * max_res), int(0.5 * max_res)))
```

Comment:
Your comment says, that the setter should not allow to set non-square resolutions. But here you check if the VA is equal to the non-square resolution. Why should the cellCompleteResolution only be squared?

Reply:
I thought we only supported square cell-images? So non square resolutions for the cell-image aren't supported and should not be allowed to be set.

Reply:
There is no special setter for the VA - so nothing that checks for square resolution. So the VA should allow non-square cells. Your test is actually checking for a non-square resolution. I checked the API and also the specs and couldn't find that they cannot be non-square. Did I miss it somewhere? 
What we do not support is non-square pixels!

Reply:
TODO TODAY

Reply:
No you are right I was confused with the pixel size.
I have updated the comment and the test.